hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceFactory.java

```diff
@@ -18,12 +18,14 @@
 
 import com.hazelcast.config.Config;
 import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.spi.annotation.PrivateApi;
 
 /***
  * This is interface which provides capability for Hazelcast factories customization;
  * DefaultHazelcastInstanceFactory is used by default, but external service can have;
  * custom implementation which should be specified in META-INF services;
  */
+@PrivateApi
```

Comment:
It is private since it points to NodeContext and NodeContext points to NodeExtension and NodeExtension is without any question private.